lib/ops_test.sh

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env bash
+
+# mock `cloudsmith` binary
+#
+# This prevents the actual `cloudsmith` binary from being executed during
+# these tests. Every invocation is logged to a file.
+#
+# To make assertions, simply inspect the contents of the file to
+# ensure that the expected commands _would_ have been invoked.
+
+cloudsmith() {
```

Comment:
so it looks like we're stubbing all possible inputs/outputs from cloudsmith (at test time) all in one go; is there a way to do that at each test instead? so each input/output is associated with its test function

if not it's not a big deal, like, i know it's bash

Reply:
I think this is the most straightforward way at the moment ☹️